### PR TITLE
Feature/simple new functionality

### PR DESCRIPTION
**Pull request by pani-tka**
Editing functions completed

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -26,11 +31,25 @@ export default {
 
   methods: {
     addTodo (newTodo) {
-      this.todos.push(newTodo)
+      let todo = {
+        title: newTodo,
+        completed: false,
+        editing: false
```
Comment (hollote) [inline src/App.vue]: you don't need `completed: false` and `editing: false` for now

File: src/App.vue [hunk 2]
```diff
@@ -26,11 +31,25 @@ export default {
...
+      };
+      if (newTodo != '') {
```
Comment (hollote) [inline src/App.vue]: please do not send `add event` if the title is empty

File: src/App.vue [hunk 3]
```diff
@@ -26,11 +31,25 @@ export default {
...
+        this.todos.push(todo)
+      } 
     },
 
     removeTodoById (id) {
       this.todos.splice(id, 1);
+    },
+
+    editTodo (index, todo) {
+      if(todo.title != '') {
```
Comment (hollote) [inline src/App.vue]: if title empty -> return;

File: src/App.vue [hunk 4]
```diff
@@ -26,11 +31,25 @@ export default {
...
+        if (todo.title.trim() == "") todo.title = this.editTodoCache;
```
Comment (hollote) [inline src/App.vue]: add `{}`brackets to if condition

File: src/components/add-new-todo.vue [hunk 5]
```diff
@@ -18,7 +18,7 @@ export default Vue.component('AddNewTodo', {
     clearTodoInput () {
```
Comment (hollote) [inline src/components/add-new-todo.vue]: change naming to correct one

File: src/components/todo-item.vue [hunk 6]
```diff
@@ -1,7 +1,20 @@
 <template>
   <li>
-    <label>{{todo}}</label>
-    <button @click="$emit('removeTodoItem')">Remove</button>
+    <div v-if="!todo.editing" @click="editMode(todo)">
```
Comment (hollote) [inline src/components/todo-item.vue]: u can use `isEditing` condition without item attribute (it is redundant)

File: src/components/todo-item.vue [hunk 7]
```diff
@@ -1,7 +1,20 @@
...
+      {{todo.title}} 
+      <button @click="$emit('editTodo',editTodoCache)">Edit</button>
```
Comment (hollote) [inline src/components/todo-item.vue]: editTodoCache <- please change naming to correct one

Comment (hollote) [inline src/components/todo-item.vue]: Do you really need to trigger edit on start editing?

File: src/components/todo-item.vue [hunk 8]
```diff
@@ -10,7 +23,32 @@ import Vue from 'vue'
...
+
+  data() {
+    return {
+      editTodoCache: ""
+    }
+  },
+
+  methods:{
+    editMode (todo) {
+      todo.editing = true;
```
Comment (hollote) [inline src/components/todo-item.vue]: You don't need `todo.editing` here

File: src/App.vue (hunk 1 above)
Comment (pani-tka) [reply src/App.vue]: Removed from todo.

File: src/App.vue (hunk 3 above)
Comment (pani-tka) [reply src/App.vue]: Fixed.

File: src/App.vue (hunk 4 above)
Comment (pani-tka) [reply src/App.vue]: Fixed.

File: src/App.vue (hunk 2 above)
Comment (pani-tka) [reply src/App.vue]: Fixed. Check moved to AddNewTod-component.

File: src/components/add-new-todo.vue (hunk 5 above)
Comment (pani-tka) [reply src/components/add-new-todo.vue]: clearTodoInput()  changed to createNewTodo().

File: src/components/todo-item.vue (hunk 6 above)
Comment (pani-tka) [reply src/components/todo-item.vue]: Fixed.

File: src/components/todo-item.vue (hunk 7 above)
Comment (pani-tka) [reply src/components/todo-item.vue]: editTodoCache -> beforeEditingValue

Comment (pani-tka) [reply src/components/todo-item.vue]: Fixed.

File: src/components/todo-item.vue (hunk 8 above)
Comment (pani-tka) [reply src/components/todo-item.vue]: Removed.

File: src/App.vue [hunk 9]
```diff
@@ -26,11 +31,16 @@ export default {
...
-      this.todos.push(newTodo)
+      let todo = {title: newTodo};
+      this.todos.push(todo)
     },
-
     removeTodoById (id) {
       this.todos.splice(id, 1);
+    },
+    editTodo (index, todo) {
+      if (todo.title.trim() == "") {
```
Comment (hollote) [inline src/App.vue]: Where do you use index?

File: src/components/todo-item.vue [hunk 10]
```diff
@@ -10,7 +23,39 @@ import Vue from 'vue'
...
+        this.isEditing = false;
+      }
+    },
+    cancelEdit () {
+      this.todo.title = this.beforeEditingValue;
+      this.isEditing = false;
+    }
+  },
+
+  directives: {
```
Comment (hollote) [inline src/components/todo-item.vue]: WAT?

File: src/components/todo-item.vue [hunk 11]
```diff
@@ -10,7 +23,39 @@ import Vue from 'vue'
...
+      this.beforeEditingValue = this.todo.title;
+    },
+    saveEdit () {
+      if (this.todo.title != '') {
+        this.$emit('editTodo', this.beforeEditingValue);
+        this.isEditing = false;
+      }
+    },
+    cancelEdit () {
+      this.todo.title = this.beforeEditingValue;
```
Comment (hollote) [inline src/components/todo-item.vue]: do we need this?

Comment (hollote) [reply src/components/todo-item.vue]: props are immutable!

File: src/components/todo-item.vue [hunk 12]
```diff
@@ -10,7 +23,39 @@ import Vue from 'vue'
...
 export default Vue.component('TodoItem', {
   props: {
-    todo: String,
+    todo: Object,
+  },
+
+  data() {
+    return {
+      isEditing: false,
+      beforeEditingValue: ""
```
Comment (hollote) [inline src/components/todo-item.vue]: wrong name and usage

File: src/components/todo-list.vue [hunk 13]
```diff
@@ -5,6 +5,7 @@
       :key="index"
       :todo="todo"
       @removeTodoItem="$emit('removeTodoById', index)"
+      @editTodo="$emit('editTodo', index, todo)"
```
Comment (hollote) [inline src/components/todo-list.vue]: todo? we just need title here

File: src/App.vue [hunk 14]
```diff
@@ -26,11 +31,16 @@ export default {
 
   methods: {
     addTodo (newTodo) {
-      this.todos.push(newTodo)
+      let todo = {title: newTodo};
```
Comment (hollote) [inline src/App.vue]: please change object to plain text

**Pull request merged**